index.tt

```diff
@@ -6,152 +6,349 @@
+          application needs.
+        </p>
+        <p>
+          The following Nix expression (<code>default.nix</code>) defines a
+          docker image with <strong>only</strong> Python 3 installed in it.
+        </p>
+        <pre class="well" style="margin-top:1em">
+{ pkgs ? import &lt;nixpkgs&gt; {}
+}:
+pkgs.dockerTools.buildLayeredImage {
+  name = "only-python";
+  contents = [ pkgs.python3 ];
+}</pre>
+        <p>To build and run the image you need to:</p>
+        <pre class="well" style="margin-top:1em">
+<span class="nix-command"><strong>nix-build</strong></span>
+...
+/nix/store/hb8xwmgd1pfrwgmc3indi2rrgcjr3wa3-docker-image-only-python.tar.gz
+<span class="nix-command">docker load -u ./result</span>
+...
+Loaded image: hello:hb8xwmgd1pfrwgmc3indi2rrgcjr3wa3
+<span class="nix-command">docker run hello:hb8xwmgd1pfrwgmc3indi2rrgcjr3wa3 python -c "print('hello world')"</span>
+hello world</pre>
+        <p>
+          Learn more <a href="[%root%]nixpkgs/manual/#sec-pkgs-dockerTools">how
+            to build docker images</a>.
+        </p>
+
+      </div>
+      <div class="span6">
+
+        <h3>Declarative cloud images</h3>
+        <p>How hard would it be to build and configure <strong>Amazon AMI?</strong></p>
+        <p>
+          With the following <code>amazon.nix</code> we defined nginx which is
+          serving example <code>/var/www</code> folder, having a valid ssl
+          certificate (via LetsEncrypt) and enabled recommended security
+          settings.
+        </p>
+        <pre class="well" style="margin-top:1em">
+{ pkgs, ...}:
+{
+  security.acme.acceptTerms = true;
+  security.acme.email = "nix@example.com";
+  services.nginx = {
+    enable = true;
+    recommendedGzipSettings = true;
+    recommendedOptimisation = true;
+    recommendedProxySettings = true;
+    recommendedTlsSettings = true;
+    virtualHosts."example.com" = {
+      enableACME = true;
+      forceSSL = true;
+      locations."/".root = "/var/www";
+    };
+  };
+}</pre>
+        <p>Now we just need to build it and upload it.</p>
+        <pre class="well" style="margin-top:1em">
+<span class="nix-command"><strong>nix-build '&lt;nixpkgs/nixos/release.nix&gt;' -A amazonImage.x86_64-linux --arg configuration ./amazon.nix</strong></span>
+...
+/nix/store/wmg6gcgnh6bk8d98syydfq73q6hjv682-nixos-amazon-image-20.09pre130979.gfedcba-x86_64-linux
+<span class="nix-command">ls -lh ./result/</span>
+-r--r--r--     1 root root   1.4G Jan  1  1970 nixos-amazon-image-20.09pre130979.gfedcba-x86_64-linux.vhd
+dr-xr-xr-x     2 root root      4 Jan  1  1970 nix-support
+<span class="nix-command"><strong>nix-shell -p awscli</strong></span>
+(nix-shell)<span class="nix-command">export AWS_IMAGE="./result/nixos-amazon-image-20.09pre130979.gfedcba-x86_64-linux.vhd"</span>
+(nix-shell)<span class="nix-command">aws s3 cp --region $REGION $AWS_IMAGE "S3://${BUCKET}/${AWS_IMAGE#/}"</span>
```

Comment:
explain where `$REGION` comes from

copying is fine, but they should also run it and see the website

Reply:
For now i will only write how to build it. we can later improve on this